### PR TITLE
solution

### PR DESCRIPTION
**Pull request by lukacsja**
[DEMO LINK](https://lukacsja.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -15,48 +16,113 @@ export const goodsFromServer = [
...
           Reverse
         </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(isReversed || sortType !== NONE) && (
```
Comment (kenvastolord) [inline src/App.tsx:105]: I would recommend save (isReversed || sortType !== NONE) into a variable and then use the variable here, just that, everything from that point looks wonderful.

**Review by kenvastolord — COMMENTED** [review]
Excellent job on that project.

File: src/App.tsx [hunk 2]
```diff
@@ -15,48 +16,113 @@ export const goodsFromServer = [
...
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {
+          visibleGoods.map(good => (
+            <li data-cy="Good" key={good}>
```
Comment (naziamoff) [inline src/App.tsx:121]:
```suggestion
        {visibleGoods.map(good => (
            <li data-cy="Good" key={good}>
```

File: src/App.tsx [hunk 3]
```diff
@@ -15,48 +16,113 @@ export const goodsFromServer = [
...
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setReverseState] = useState(false);
```
Comment (naziamoff) [inline src/App.tsx:56]: no need to think of the names for states. It always works like `[value, setValue]`
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```

**Review by naziamoff — APPROVED** [review]
well done, but I recommend you to fix a couple of things including the @kenvastolord comment

**Pull request closed without merging**